x-pack/plugins/beats_management/public/components/table/controls/option_control.tsx

```diff
@@ -4,193 +4,54 @@
+  switch (props.type) {
+    case ActionComponentType.Action:
+      if (!props.action) {
+        throw Error('Action cannot be undefined');
       }
-    }
-
-    panels.push(panel);
-
-    if (schemaArray !== null) {
-      schemaArray.forEach((item: AssignmentControlSchema) => {
-        if (item.panel) {
-          this.schemaToPanelTree(item.panel, panels);
-        }
-      });
-    }
-
-    return panels;
-  }
-
-  public render() {
-    const { itemType, selectionCount, schema, intl } = this.props;
-
-    return (
-      <EuiPopover
-        button={
-          <FixedEuiToolTip
-            position="top"
-            delay="long"
-            content={
-              selectionCount === 0
-                ? intl.formatMessage(
-                    {
-                      id: 'xpack.beatsManagement.tableControls.selectItemDescription',
-                      defaultMessage:
-                        'Select {itemType} to perform operations such as setting tags and unenrolling Beats.',
-                    },
-                    { itemType }
-                  )
-                : intl.formatMessage(
-                    {
-                      id: 'xpack.beatsManagement.tableControls.manageSelectedItemDescription',
-                      defaultMessage: 'Manage your selected {itemType}',
-                    },
-                    { itemType }
-                  )
-            }
-          >
-            <EuiButton
-              color="primary"
-              iconSide="right"
-              disabled={selectionCount === 0}
-              iconType="arrowDown"
-              onClick={() => {
-                this.setState({
-                  showPopover: true,
-                });
-              }}
-            >
-              <FormattedMessage
-                id="xpack.beatsManagement.tableControls.manageItemButtonLabel"
-                defaultMessage="Manage {itemType}"
-                values={{ itemType }}
-              />
-            </EuiButton>
-          </FixedEuiToolTip>
-        }
-        closePopover={() => {
-          this.setState({ showPopover: false });
-        }}
-        id="assignmentList"
-        isOpen={this.state.showPopover}
-        panelPaddingSize="none"
-        anchorPosition="downLeft"
-        withTitle
-      >
-        <EuiContextMenu initialPanelId={0} panels={this.schemaToPanelTree(schema)} />
-      </EuiPopover>
-    );
+      return (
+        <ActionControl
+          actionHandler={props.actionHandler}
+          action={props.action}
+          danger={props.danger}
+          name={props.name}
+          showWarning={props.showWarning}
+          warningHeading={props.warningHeading}
+          warningMessage={props.warningMessage}
+          disabled={props.disabled}
+        />
+      );
+    case ActionComponentType.TagBadgeList:
+      if (!props.actionDataKey) {
+        throw Error('actionDataKey cannot be undefined');
```

Comment:
This is a dev only error... and only exists because I was too lazy to type this correctly with TS... so maybe that's the bug? Ill leave it up to you if you want to LGTM with that knowledge or want me to fix it 😊

Reply:
no worries then